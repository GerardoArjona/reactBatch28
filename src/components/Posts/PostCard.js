import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export default class PostCard extends Component{
    render(){
        return(
            <div className="card">
                <div className="card-title">
                    <Link to={`/posts/${this.props.id}`}>
                        <h1>{this.props.title}</h1>
                    </Link>
                </div>
                <div className="card-content">
                    <span className="card-text">
                        {this.props.first_name}
                    </span>
                </div>
            </div>
        );
    }
}