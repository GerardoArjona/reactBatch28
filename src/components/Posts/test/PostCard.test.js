import React from 'react';
import { shallow } from "enzyme";

import '../../../setupTest';

import PostCard from '../PostCard';

describe("<PostCard />", () => {
    it("Debe ejecutar el render conrrectamente", ()=>{
        const component = shallow(<PostCard/>);
        expect(component).toMatchSnapshot();
    })

    it("Debe pintar props", () => {
        const post = {
            id:"pruebaid",
            title:"Prueba post",
            author:{
                first_name:"Gerardo"
            }
        }

        const component = shallow(<PostCard
            id={post.id}
            title={post.title}
            first_name={post.author.first_name}
        />)

        expect(component.find(".card-text").text()).toBe(post.author.first_name)
    })
})