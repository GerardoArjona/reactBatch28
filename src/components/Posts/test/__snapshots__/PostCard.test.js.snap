// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Debe ejecutar el render conrrectamente 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
        >
          <Link
            replace={false}
            to="/posts/undefined"
          >
            <h1 />
          </Link>
        </div>,
        <div
          className="card-content"
        >
          <span
            className="card-text"
          />
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/posts/undefined"
          >
            <h1 />
          </Link>,
          "className": "card-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1 />,
            "replace": false,
            "to": "/posts/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="card-text"
          />,
          "className": "card-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-text",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
          >
            <Link
              replace={false}
              to="/posts/undefined"
            >
              <h1 />
            </Link>
          </div>,
          <div
            className="card-content"
          >
            <span
              className="card-text"
            />
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/posts/undefined"
            >
              <h1 />
            </Link>,
            "className": "card-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h1 />,
              "replace": false,
              "to": "/posts/undefined",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="card-text"
            />,
            "className": "card-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card-text",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
